frontend/libs/ui/pages/src/lib/calculator/boiler/boiler.component.ts

```diff
@@ -6,42 +6,59 @@ import { Boiler, EnergyLabels, Location } from '@lectoraat-smart-energy/shared';
 import { BoilerResultComponent } from '@lectoraat-smart-energy/ui/components';
 import { Observable } from 'rxjs';
 import { LocationService } from '../../location/location.service';
+import { ActivatedRoute, RouterModule } from '@angular/router';
+
 
 @Component({
   selector: 'lectoraat-smart-energy-boiler',
   standalone: true,
-  imports: [CommonModule, FormsModule, BoilerResultComponent],
+  imports: [CommonModule, FormsModule, BoilerResultComponent, RouterModule],
   providers: [BoilerService, LocationService],
   templateUrl: './boiler.component.html',
   styleUrl: './boiler.component.css',
 })
 export class BoilerComponent implements OnInit {
-  boiler: Boiler = {} as Boiler;
+  boiler!: Boiler;
   locations$!: Observable<Location[]>;
+
   energyLabels = Object.keys(EnergyLabels);
 
   energyLabelColors: { [key: string]: { color: string } } = {
-    'a+': { color: 'green' },
-    a: { color: '#FFD700' }, //yellow was too light, darkyellow too dark so the hexcode #FFD700 will be used.
-    b: { color: 'orange' },
-    c: { color: 'red' },
-    d: { color: 'purple' },
-    e: { color: 'blue' },
-    f: { color: 'indigo' },
-    g: { color: 'violet' },
-  };
+      'a+': { color: 'green' },
+      a: { color: '#FFD700' }, //yellow was too light, darkyellow too dark so the hexcode #FFD700 will be used.
+      b: { color: 'orange' },
+      c: { color: 'red' },
+      d: { color: 'purple' },
+      e: { color: 'blue' },
+      f: { color: 'indigo' },
+      g: { color: 'violet' },
+    };
 
   constructor(
     private boilerService: BoilerService,
     private locationService: LocationService,
+    private route: ActivatedRoute,
   ) {}
 
   ngOnInit(): void {
+    const id = this.route.snapshot.paramMap.get('id') ?? '';
+
+    if (id) {
+      this.boilerService.getBoiler(id).subscribe((boiler) => {
+        this.boiler = boiler;
+      });
+    } else {
+      this.boiler = {} as Boiler;
```

Comment:
Zou deze definitie bovenin de class doen, dan kun je de else weg halen. Zorgt er voor dat je in de template ook niet hoeft af te vangen dat de boiler `null` kan zijn